### ISSUE TITLE
Retain order of settings after sync

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#85716]
People who manage settings by editing the file directly often order and organize it specially, ideally we should try merge into the right place and retain the original order.

**Comment by Tyriar** [comment]
Another issue with this is comments, I don't use them in my settings file but I know some people comment sections. For me I will comment out a setting temporarily (eg. `terminal.integrated.rendererType`), if I then change the setting on another machine, the original machine will end up having the comment and the setting.

**Comment by sandy081** [comment]
Order and comments are not considered in favour of providing better merge while synchronising. Otherwise user has to manually merge most of the times. Rich Settings UI is also another reason for this.

I will think about this If there is more demand for this.

**Comment by Tyriar** [comment]
@sandy081 I see plenty of settings files organized like mine when I ask people for settings. Is the problem implementing the merge algorithm, perhaps we could leverage git if it's available to do the merge for us?

**Comment by sandy081** [comment]
git is not available as it is done in core. we have to implement 3-way merge for text or do not merge and let the user do it by showing the diff of remote and local.

**Comment by Tyriar** [comment]
This is still a big annoyance for me, I went from a perfectly organized settings file everywhere before settings sync (syncing with dotfiles) to a mess that's different on every machine.

**Comment by sandy081** [comment]
If there is a setting update, then it is updated at the same place. Only additions are added at the bottom. Any idea how your settings got dis-organised for you?

**Comment by Tyriar** [comment]
> Only additions are added at the bottom.

This is the problem, all new settings will only be organized on the machine in which they were added. It's mostly in order for my work desktop as that's where I make most of my settings changes, however on my macbook the bottom of my settings file looks like this:

```
"workbench.dialogs.useSimplified": "true",
"editor.folding": false,
"errorLens.errorForeground": "#da6771",
"errorLens.errorBackground": "#da677140",
"errorLens.warningForeground": "#cca700",
"errorLens.warningBackground": "#cca70040",
(snip)
(snip)
"terminal.integrated.rendererType": "experimentalWebgl",
"window.zoomLevel": 0
```

Just a mess of settings that have been introduced since. On my home desktop it's even worse as I change less there.

**Comment by chrisdias** [comment]
I have the same problem. I edit my `settings.json` manually and have it organized in sections with comments and blank lines. In the sync'd settings it is one block of `json`.  I am happy that the order is the same, I just want my comments and blank lines too. 

![image](https://user-images.githubusercontent.com/1487073/72180029-18f6db00-339b-11ea-9baf-c314d4f51f03.png)

I will admit that I originally organized my settings this way for presentations, but I've kept it that way because I find it easy to find the settings that I've set and want to tweak.
